presto-hive/src/main/java/com/facebook/presto/hive/orc/OrcSelectivePageSourceFactory.java

```diff
@@ -640,13 +640,13 @@ private static OrcPredicate toOrcPredicate(TupleDomain<Subfield> domainPredicate
         filter = and(extractDynamicFilterResult.getStaticConjuncts());
 
         if (!isAdaptiveFilterReorderingEnabled(session)) {
-            filterFunctions.add(new FilterFunction(session.getSqlFunctionProperties(), determinismEvaluator.isDeterministic(filter), predicateCompiler.compilePredicate(session.getSqlFunctionProperties(), filter).get()));
+            filterFunctions.add(new FilterFunction(session.getSqlFunctionProperties(), determinismEvaluator.isDeterministic(filter), predicateCompiler.compilePredicate(session.getSqlFunctionProperties(), session.getSessionFunctions(), filter).get()));
```

Comment:
Can we make it a single api to get all visible functions instead of adding the second param through out?

Reply:
I did some refactoring and reduced the "blast radius" of adding this second param and reduced the size of this PR by about 50 files. There's no easier solution here, because most of the compilers are singletons, connectors can access the compilers, the connectors don't have session, they have connector session, and so it all has to get threaded through.